frontend/src/components/Playback/MatchingPairs.js

```diff
@@ -100,20 +98,8 @@ const MatchingPairs = ({
                 currentCard.noevents = true;
                 // check for match
                 const lastCard = sections[firstCard.current];
-                if (lastCard.group === currentCard.group) {
-                    // match
-                    if (currentCard.seen) {
-                        score.current = 20;      
-                        setTurnFeedback('+1')
-                    } else {
-                        score.current = 10;
-                        setTurnFeedback('0')
-                    }
-                } else {
-                    if (currentCard.seen) { score.current = -10; }
-                    else { score.current = 0; }
-                    setTurnFeedback('-1')
-                };
+                const imScore = await scoreIntermediateResult({session, participant, result: {currentCard, lastCard}});
+                score.current = imScore.score;
```

Comment:
if the http request of `scoreIntermediateResult` fails, it returns a `null` value. When that happens you cannot do `score.current = imScore.score;`. It might block the application, or not?

Reply:
Good point!